README.md

```diff
@@ -1,3 +1,84 @@
 # Laravel gitlab deploy
 
-Prepare your project to deploy
+This package created to optimize and improve company CI/CD in Gitlab environment.
+
+# Requirements
+
+This package requires **PHP 7.4**  and **Laravel 8.0**.
+
+Available SSH agents on local machine and remote server.
+
+Remote server must use GNU/Linux
+
+# Installation
+
+You can install this package via composer using:
+
+```shell
+composer required --dev hexide-digital/hexide-admin-panel:^0.3
+```
+
+The package will automatically register its service provider.
+
+After installing, you maybe want to publish files for deploy and sample files, right? Just do this:
+
+```shell
+php artisan vendor:publish --tag="gitlab-deploy" --force
+```
+
+Source of all publishable files are copied
+from [this folder](https://github.com/hexidedigital/laravel-gitlab-deploy/tree/0.x/examples)
+to places and folder described
+in [service provider](https://github.com/hexidedigital/laravel-gitlab-deploy/blob/0.x/src/GitlabDeployServiceProvider.php#L25-L31)
+
+# Usage
+
+## Prepare
+
+Open _(after publishing)_ or create file
+[`deploy/deploy-prepare.yml`](https://github.com/hexidedigital/laravel-gitlab-deploy/blob/0.x/examples/deploy-prepare.example.yml)
+and fill all needed options:
+
+- api key for gitlab
```

Comment:
1. replace `api key` to `token` like in file
2. Describe where to get that token